File: Search/maven/src/main/java/Search/RegexOperator.java
package Search;

import java.util.ArrayList;
import java.util.regex.Pattern;
import java.util.regex.Matcher;

public class RegexOperator {

    static final String PLUS_REGEX = "(\\+)(\\w*)";
    static final String MINUS_REGEX = "(-)(\\w*)";
    static final String NONE_SIGN_REGEX = "^(\\w*)";

    public static ArrayList<String> assortTheWords(String input, String regex, int groupOfWordInRegex) {

        ArrayList<String> wordsThatMatch = new ArrayList<>();
        Pattern pattern = Pattern.compile(regex);
        String[] splitInput = input.split(" ");

        for (String s : splitInput) {
            Matcher matcher = pattern.matcher(s);
            if (matcher.matches())
                wordsThatMatch.add(matcher.group(groupOfWordInRegex));
        }
        return wordsThatMatch;
    }

}

File: Phase05/Search/Search/Model/RegexOperator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text.RegularExpressions;

namespace Search.Model
{
    public class RegexOperator
    {
         public static List<string> AssortTheWords(string input, Regex regex, int groupOfWordInRegex)
         {
            List<string> wordsThatMatch = new List<string>();
             
            var matches = regex.Matches(input);
             
            if(matches == null)
            {
                throw new ArgumentNullException();
            }
            return matches.Select(m => m.Groups[2].Value).ToList();

             
         }
    }
}